tee-worker/ts-tests/integration-tests/common/utils/crypto.ts

```diff
@@ -144,8 +145,17 @@ export class BitcoinSigner implements Signer {
     sign(message: HexString | string | Uint8Array): Promise<Uint8Array> {
         return new Promise((resolve, reject) => {
             if (isString(message)) {
-                const sig = new bitcore.Message(message).sign(this.keypair);
-                resolve(bufferToU8a(Buffer.from(sig, 'base64')));
+                let buffer: Uint8Array;
+
+                // Here we must use the format [u8,65],
+                // so we can only accept a signature that has been decoded from base64 and has a length of 65.
+                // Therefore, using a do -while loop to resign unnecessary signatures.We should not encounter an infinite loop situation, so no condition is set to break out of the loop.
+                do {
```

Comment:
I somehow feel this isn't the right way, still.

A btc message signature should be 65 bytes - the encoded base64 array should be 88 bytes. This should be firm. If we get less than 65 bytes in decoding, I can only think of padding?

Let it run again and hope to get 65 bytes isn't the good practice IMO, as in reality you should be able to verify each btc signature the user generated, with arbitrary `k`

cc @kziemianek  - maybe you know it as you've done some btc sig related impl

Reply:
In my opinion it should always produce 65 bytes signature, even `bitcore-lib` developers assumes that. 
If you try to use `bitcode-lib` api to recreate signature from `Buffer` it will fail:

```
const buffer = Buffer.from(sig, 'base64')
Signature.fromCompact(buffer)

Invalid Argument: Error: s must be 32 bytes
Error
    at new NodeError (/home/kziemianek/projects/kziemianek/sig-test/node_modules/bitcore-lib/lib/errors/index.js:20:41)
    at Object.checkArgument (/home/kziemianek/projects/kziemianek/sig-test/node_modules/bitcore-lib/lib/util/preconditions.js:14:13)
    at Signature.fromCompact (/home/kziemianek/projects/kziemianek/sig-test/node_modules/bitcore-lib/lib/crypto/signature.js:53:5)
```

I debugged it for a while and it turns out the `r` value is sometimes 31 bytes long, which results in 64 (1+31+32) bytes signature (instead of 1+32+32). Above error says `s must be 32 bytes` but the root cause is still `r` being 31 bytes long.

![64-byte-sig](https://github.com/litentry/litentry-parachain/assets/1838338/556de25b-4562-473c-93e6-38c4c9746c13)

It happens ocassionally so it's really low probability of happeining twice in the row.

Reply:
So to sum up:

* switch to other js bitcoin lib (https://www.npmjs.com/package/bitcoinjs-lib maybe ?)
* implement a workaround (try to generate new signature untill it has 65 bytes) and wait for the potential fix in `bitcore-lib` - but before that happening it needs to be reported, confirmed and released.

Reply:
cc @jonalvarezz - in case you have some experience when implementing SDK..

Reply:
thanks @jonalvarezz  I actually looked at the code on idehub, and in the Bitcoin signature part, mainly use unisat-wallet. This wallet uses their own integrated [unisat-sdk](https://github.com/unisat-wallet/extension/blob/master/package.json#L54), and you will find that they also use [bitcore-lib and bitcoinjs-lib](https://github.com/unisat-wallet/wallet-sdk/blob/master/package.json#L39-L43).So, essentially there is no difference,  I guess.:p